same issue for quickbuild, must be a luarocks server issue.

```
13:25:54,311 INFO  - make[3]: Entering directory '/home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/.deps'
13:25:54,319 INFO  - [ 28%] Generating usr/lib/luarocks/rocks/mpack
13:26:43,005 INFO  - CMakeFiles/mpack.dir/build.make:60: recipe for target 'usr/lib/luarocks/rocks/mpack' failed
13:26:43,005 INFO  - make[3]: Leaving directory '/home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/.deps'
13:26:43,005 WARN  - make[3]: *** [usr/lib/luarocks/rocks/mpack] Terminated
13:26:43,007 WARN  - make[2]: *** [CMakeFiles/mpack.dir/all] Error 2
13:26:43,007 WARN  - make[1]: *** [all] Error 2
13:26:43,007 WARN  - make: *** [deps] Error 2
13:26:43,007 INFO  - CMakeFiles/Makefile2:571: recipe for target 'CMakeFiles/mpack.dir/all' failed
13:26:43,007 INFO  - make[2]: Leaving directory '/home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/.deps'
13:26:43,007 INFO  - Makefile:83: recipe for target 'all' failed
13:26:43,007 INFO  - make[1]: Leaving directory '/home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/.deps'
13:26:43,007 INFO  - Makefile:76: recipe for target 'deps' failed
13:26:54,440 INFO  - Executing post-execute action...
13:26:54,440 ERROR - Step 'master>buildall>build-node?testNode=linux-64>build-deps' is failed: Failed to run command: make deps
Command return code: 2
Command error output: Warning: variable CFLAGS was not passed in build_variables
No existing manifest. Attempting to rebuild...
ar: `u' modifier ignored since `D' is the default (see `U')
ar: `u' modifier ignored since `D' is the default (see `U')
make[3]: *** [usr/lib/luarocks/rocks/mpack] Terminated
make[2]: *** [CMakeFiles/mpack.dir/all] Error 2
make[1]: *** [all] Error 2
make: *** [deps] Error 2
```

--- a/runtime/doc/options.txt
+++ b/runtime/doc/options.txt
@@ -6922,7 +6922,8 @@ A jump table for the options with a short description can be found at |Q_op|.
 'writedelay' 'wd'	number	(default 0)
 			global
 	The number of milliseconds to wait for each character sent to the
-	screen.  When non-zero, characters are sent to the terminal one by
-	one.  For debugging purposes.
+	screen.  When positive, characters are sent to the UI one by one.
+	When negative, all redrawn characters cause a delay, even if the
+	character already was displayed by the UI.  For debugging purposes.
 
  vim:tw=78:ts=8:ft=help:noet:norl:

--- a/src/nvim/screen.c
+++ b/src/nvim/screen.c
@@ -4361,7 +4361,8 @@ static int char_needs_redraw(int off_from, int off_to, int cols)
                       && comp_char_differs(off_from, off_to))
                   || ((*mb_off2cells)(off_from, off_from + cols) > 1
                       && ScreenLines[off_from + 1]
-                      != ScreenLines[off_to + 1])))));
+                      != ScreenLines[off_to + 1])))
+          || p_wd < 0));
 }
 
 /*

--- a/src/nvim/ui.c
+++ b/src/nvim/ui.c
@@ -437,9 +437,8 @@ void ui_puts(uint8_t *str)
 
     if (p_wd) {  // 'writedelay': flush & delay each time.
       ui_flush();
-      assert(p_wd >= 0
-             && (sizeof(long) <= sizeof(uint64_t) || p_wd <= UINT64_MAX));
-      os_delay((uint64_t)p_wd, false);
+      uint64_t wd = (uint64_t)labs(p_wd);
+      os_delay(wd, false);
     }
   }
 }

